services/indexer/repo/AgencyJsonStream.js

```diff
@@ -0,0 +1,206 @@
+            .then(data => fulfill(data))
+            .catch(err => reject(`errorMessage ${agency.codeUrl} - ${err}`));
+        });
+      }
+    });
+  }
+
+  _validateAgencyRepos(agency, codeJson) {
+    logger.debug('Entered _validateAgencyRepos');
+
+    let reportDetails = [];
+    let reportString = "";
+    let totalErrors = 0;
+    let validationTotals = {
+      errors: 0,
+      warnings: 0,
+      enhancements: 0
+    };
+
+    Reporter.reportVersion(agency.acronym, codeJson.version);
+
+    let resultRepos = [];
+
+    codeJson.projects.map(repo => {
+      const repoId = Utils.transformStringToKey([agency.acronym, repo.organization, repo.name].join("_"));
+      
+      return Validator.validateRepo(repo, agency, (error, results) => {
+        if(error) {
+          logger.error(`Error validating repo with repoID ${repoId}. Throwing it out of the indexing process.`);
+        } else {
+          if(results.issues) {
+            validationTotals.errors += results.issues.errors.length ? results.issues.errors.length : 0;
+            validationTotals.warnings += results.issues.warnings.length ? results.issues.warnings.length : 0;
+            validationTotals.enhancements += results.issues.enhancements.length ?
+              results.issues.enhancements.length : 0;
+  
+            Reporter.reportIssues(agency.acronym, results);
+          }
+
+          resultRepos.push(repo);
+        }
+      });
+    });
+
+    if(validationTotals.errors) {
+      totalErrors += validationTotals.errors;
+      reportDetails.push(`${validationTotals.errors} ERRORS`);
+    }
+    if(validationTotals.warnings) {
+      totalErrors += validationTotals.warnings;
+      reportDetails.push(`${validationTotals.warnings} WARNINGS`);
+    }
+
+    if(validationTotals.enhancements) {
+      reportDetails.push(`${validationTotals.enhancements} REQUESTED ENHANCEMENTS`);
+    }
+
+    if(totalErrors) {
+      reportString= "NOT FULLY COMPLIANT: ";
+    } else {
+      agency.requirements.schemaFormat = 1;
+      reportString= "FULLY COMPLIANT: ";
+    }
+
+    reportString += reportDetails.join(", ");
+    Reporter.reportStatus(agency.acronym, reportString);
+
+    agency.requirements.overallCompliance = this._calculateOverallCompliance(agency.requirements);
+    Reporter.reportRequirements(agency.acronym, agency.requirements);
+    
+    return Promise.resolve(resultRepos);
+  }
+
+  _calculateOverallCompliance(requirements){
+    //overallCompliance should be the average of the other requirements.
+    //TODO: align this approach with project-open-data's approach
+    let overallCompliance = 0;
+    for (let req in requirements){
+      overallCompliance += requirements[req];
+    }
+    overallCompliance /= _.size(requirements);
+    return overallCompliance;
+  }
+  
+  _formatRepos(agency, repos) {
+    logger.debug('Entered _formatCodeJson - Agency: ', agency.acronym);
+
+    repos.map(repo => {
```

Comment:
We are `map`ping without a transform. It is better to use `forEach` in this case.